decredplugin/decredplugin.go

```diff
@@ -199,3 +203,170 @@ func DecodeVoteResultsReply(payload []byte) (*VoteResultsReply, error) {
 
 	return &v, nil
 }
+
+// CommentCensor is a journal entry for a censored comment.  The signature and
+// public key are from the admin that censored this comment.
+type CommentCensor struct {
+	Token     string `json:"token"`     // Censorship token
+	CommentID string `json:"commentid"` // Comment ID
+	Signature string `json:"signature"` // Admin Signature of Token+CommentID
+	PublicKey string `json:"publickey"` // Pubkey used for Signature
+}
+
+// CommentCensorReply returns the receipt for the cenosring action. The receipt
+// is the politeaid side signature of CommentCensor.Signature.
+type CommentCensorReply struct {
+	Receipt string `json:"receipt"` // Server signature of the admin Signature
+}
+
+// Comment is the structure that describes the full server side content.  It
+// includes server side meta-data as well. Note that the receipt is the server side
+type Comment struct {
```

Comment:
Don't we need a field `userid` within the Comment struct so we can infer the username when replying to the client? (so we can display the author of the comment).